### DISCUSSION TITLE
Nginx Proxy Manager 502 Bad Gateway

### DISCUSSION THREAD

**Discussion by danielmochalov27** [kimai/kimai#3319] [On-Premise: Installation & Updates]
Hello I have installed kimai in docker its on port 8083. I'm able to access it when I make a ssh tunnel from my pc to the server. When I setup and point nginx proxy manager to 127.0.0.1:8083 it gives me 502 Bad Gateway. Configuration for npm attached. Thank you for any help.
<img width="498" alt="Screen Shot 2022-05-22 at 1 20 42 PM" src="https://user-images.githubusercontent.com/40433547/169714360-ebdec9de-5a7d-4bc0-b21c-4d68d08a61c0.png">

<img width="469" alt="Screen Shot 2022-05-22 at 1 21 01 PM" src="https://user-images.githubusercontent.com/40433547/169714364-c0220084-2f09-43cf-8e28-b7859d9fbb64.png">

**Comment by kevinpapst** [comment]
https://github.com/kevinpapst/kimai2/discussions/3266

**Reply by danielmochalov27** [reply]
Hey so when I add 
"listen 443 ssl;
    server_name                 kimai2.local;
    ssl_certificate             /Users/kevin/Sites/_certs/kimai2.local.crt;
    ssl_certificate_key         /Users/kevin/Sites/_certs/kimai2.local.key;
    ssl_session_timeout         5m;
    ssl_protocols               SSLv2 SSLv3 TLSv1;
    ssl_ciphers                 HIGH:!aNULL:!MD5;
    ssl_prefer_server_ciphers   on;"

it disables that configuration and refuses to work any ideas?

**Reply by danielmochalov27** [reply]
I also found this out in docker logs 
""GET / HTTP/1.1" 302 562 "-" "curl/7.64.0"
any ideas anyone?

**Reply by IkerSolozabal** [reply]
Any news on what might be happening!? I have the same problem and I have not been able to find out why.

**Comment by IkerSolozabal** [comment]
# "Bad Gateway" Error because 2 instances of kimai on same server

When setting up two instances of Kimai on the same server, you may encounter the "Bad Gateway" error. This issue typically occurs when trying to run multiple containers with the same configuration or name, causing confusion in proxy redirection.

## Cause of the Issue:
In my case, the issue was caused by restoring a backup that created an additional directory, which resulted in the creation of two instances of Kimai. This duplication caused problems when Nginx or Docker tried to manage the containers, leading to a "Bad Gateway" error.

## Possible Cause:
This error may be due to attempting to run multiple instances of Kimai on the same server, but Nginx or Docker cannot manage the containers correctly if they have duplicate names or configurations. When a `proxy_pass` is made to a container and there are multiple containers with the same name or identifier, the system doesn't know which container to direct requests to, causing the "Bad Gateway" error.

## Solution:

If you want to have multiple instances of Kimai on the same server, the solution is to:
1. **Stop one instance**: If you have duplicate containers, stop one of them and remove the associated containers.
2. **Clean and remove unnecessary containers**: Remove any unnecessary containers to ensure Nginx or Docker doesn't get confused with the active instances.

**Reply by IkerSolozabal** [reply]
I got the answer from https://github.com/kimai/kimai/issues/1367
